index.rst

```diff
@@ -1,61 +1,311 @@
+The command payload would minimally be;
+
+  - RA: Right ascension of the field (in hours).
+  - Dec: Declination of the field (in degrees).
+  - PA: Position angle (in degrees).
+  - mode: An enumeration specifying the wfs mode; corner, full array, etc.
+
+Issuing the command would look something like:
+
+.. code-block:: py
+
+  from lsst.ts.idl.MTAOS import WFSMode
+
+  # Select sources for Main Camera Corner wavefront sensor mode
+  await mtaos.cmd_selectSources.set_start(ra=14., dec=-10., pa=0., type=WFSMode.Corner)
+
+  # Select sources for Main Camera Full array mode
+  await mtaos.cmd_selectSources.set_start(ra=14., dec=-10., pa=0., type=WFSMode.FullArray)
+
+  # Select sources for ComCam mode
+  await mtaos.cmd_selectSources.set_start(ra=14., dec=-10., pa=0., type=WFSMode.ComCam)
+
+Pre-processing frames
+---------------------
+
+While obtaining Intra/Extra focal images for ComCam or Main Camera full array, one would have the first image ready while the second is exposing.
+That means, it would be possible to get ahead and start the pre-processing of those frames (e.g. ISR).
+
+This could be accomplished by adding a ``preProcess`` command to the MTAOS, which would take care of that initial step.
+The command would only need to know the ``dataId`` of the frame to process, e.g.;
+
+.. code-block:: py
+
+  await mtaos.cmd_preProcess.set_start(dataId=2020103000040)
+
+With all this considered, an Intra/Extra focus sequence with ``ComCam`` could be done like:
+
+.. code-block:: py
+
+  import asyncio
+  from lsst.ts import salobj
+  from lsst.ts.observatory.control.maintel import MTCS, ComCam
+
+  domain = salobj.Domain()
+
+  mtcs = MTCS(domain=domain)
+  comcam = ComCam(domain=domain)
+
+  asyncio.gather(mtcs.start_task, comcam.start_task)
+
+  # slew/track target
+  await mtcs.slew_icrs(ra=14., dec=-10., rot=0.)
+
+  # Select sources for processing in the background
+  select_sources_task = asyncio.create_task(
+    mtcs.rem.mtaos.cmd_selectSources.set_start(
+        ra=14.,
+        dec=-10.,
+        pa=0.,
+        type=WFSMode.ComCam
+      )
+    )
+
+  # piston camera hexapod
+  await mtcs.rem.hexapod_1.cmd_move.set_start(z=1000.)
+
+  # take first exposure
+  data_id_1 = await comcam.take_eng(exptime=30)
+
+  # piston hexapod
+  await mtcs.rem.hexapod_1.cmd_move.set_start(z=-1000.)
+
+  # tell MTAOS to start processing first image in the background
+  preproc_task_1 = asyncio.create_task(mtcs.rem.mtaos.cmd_preProcess.set_start(dataId=data_id_1))
+
+  # take second exposure
+  data_id_2 = await comcam.take_eng(exptime=30)
+
+  # tell MTAOS to start processing second image in the background
+  preproc_task_2 = asyncio.create_task(mtcs.rem.mtaos.cmd_preProcess.set_start(dataId=data_id_2))
+
+  # wait for background processes to finish
+  await asyncio.gather(select_sources_task, preproc_task_1, preproc_task_2)
+
+  # Process full array intra/extra focal images
+  await mtcs.rem.mtaos.cmd_runWEP.set_start(
```

Comment:
The problem here is what is the responsibility of `runWEP` here. The function name is unclear to me.

Reply:
Not sure what you mean. I think it is reasonable to assume everyone running this understands what WEP is...

Reply:
I mean, instead of the name of `runWEP`, it might be better to have a clear name of commnads/functions. You could see for the xml we are using now, Chris tried to clarify the name of each function (although some of them are not so good). In the proposed xml here, it will look like: `selectSources()` --> `preProcess()` --> `runWEP`. But I will prefer to do: `selectSources()` --> `preProcess()` --> `calcWfErr()` or something like this.

Reply:
yes, and we should document what it does when we have a clear picture of it in the CSC documentation.